tests/integrations/server/kv_service.rs

```diff
@@ -2759,3 +2759,119 @@ fn test_pessimistic_lock_execution_tracking() {
 
     handle.join().unwrap();
 }
+
+#[test_case(test_raftstore::must_new_cluster_and_kv_client)]
+#[test_case(test_raftstore_v2::must_new_cluster_and_kv_client)]
+fn test_mvcc_scan_memory_and_cf_locks() {
+    let (_cluster, client, ctx) = new_cluster();
+    let format_key = |prefix: char, i: usize| format!("{}{:04}", prefix, i).as_bytes().to_vec();
+
+    // Create both pessimistic and prewrite locks.
+    let start_ts = 11;
+    let num_keys = 1000;
+    let prewrite_primary_key = format_key('k', 1);
+    let val = b"value";
+    for i in 0..num_keys {
+        let key = format_key('k', i);
+        if i % 2 == 0 {
+            must_kv_pessimistic_lock(&client, ctx.clone(), key, start_ts);
```

Comment:
Increasing the `in-memory` lock size limit may help I think.

Reply:
I mean, can we try to check whether this test code writes pessimistic locks to memory or storage? In case there are any missing configuration which leads to the locks still being written to storage, then the test would become ineffective

Reply:
There seems no direct way to check the memory lock status of the started cluster, some engine checks are added at L2804 to verify if the locks exist in the lock cf, PTAL.